src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -232,41 +230,53 @@ public override Task<bool> ReadAsync(CancellationToken cancellationToken)
 
         async Task<bool> Read(bool async, CancellationToken cancellationToken = default)
         {
-            switch (State)
-            {
-            case ReaderState.BeforeResult:
-                // First Read() after NextResult. Data row has already been processed.
-                State = ReaderState.InResult;
-                return true;
+            CancellationTokenRegistration? registration = null;
```

Comment:
Too bad we can't lift all this cancellation token up to ReadAsync (because it isn't async). But doesn't matter.

Reply:
:sob: